src/Features/Core/Portable/Completion/Providers/Snippets/SnippetCompletionItem.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers.Snippets
+{
+    internal class SnippetCompletionItem
+    {
+        public static CompletionItem Create(
+            string displayText,
+            string displayTextSuffix,
+            int line,
+            SyntaxToken token,
+            Glyph glyph)
+        {
+            var props = ImmutableDictionary<string, string>.Empty
+                .Add("Line", line.ToString())
+                .Add("TokenSpanStart", token.Span.Start.ToString())
+                .Add("TokenSpanEnd", token.Span.End.ToString());
+
+            return CommonCompletionItem.Create(
+                displayText: displayText,
+                displayTextSuffix: displayTextSuffix,
+                glyph: glyph,
+                properties: props,
+                isComplexTextEdit: true,
+                rules: CompletionItemRules.Default);
+        }
+
+        public static int GetTokenSpanStart(CompletionItem item)
+        {
+            if (item.Properties.TryGetValue("TokenSpanStart", out var text)
+                && int.TryParse(text, out var number))
+            {
+                return number;
+            }
+
+            return 0;
```

Comment:
what does thsi mean?  when does this happen? do you have tests for this?

Reply:
I don't think it should ever happen. Would it be better to throw an exception at that point?